src/Adapters/Storage/Mongo/MongoTransform.js

```diff
@@ -508,7 +509,14 @@ function transformConstraint(constraint, inArray) {
   if (typeof constraint !== 'object' || !constraint) {
     return CannotTransform;
   }
-
+  const transformFunction = inArray ? transformInteriorAtom : transformTopLevelAtom;
+  const transformer = (atom) => {
+    const result = transformFunction(atom);
```

Comment:
It could be re-written:

```
let transformFunction;
if (inArray) {
  transformFunction = transformInteriorAtom;
} else {
  transformFunction  = transformTopLevelAtom;
}

// transformFunction is a function that takes an atom.

```